packages/mongo-nav/src/user-menu/UserMenu.spec.tsx

```diff
@@ -54,26 +54,20 @@ describe('packages/mongo-nav/user-menu', () => {
   });
 
   test('renders university MenuItems when university dropdown is clicked and closes other SubMenus', () => {
-    const { getByTestId } = renderUserMenu({
+    const { getByText, getByTestId, getAllByRole } = renderUserMenu({
       activePlatform: 'cloud',
     });
     const trigger = getByTestId('user-menu-trigger');
     fireEvent.click(trigger);
 
-    const university = document.querySelectorAll(
-      '[data-leafygreen-ui="sub-menu-container"]',
-    )[1];
-
-    const universityArrowButton = university?.parentNode?.querySelector(
-      'button',
-    );
+    const universityTrigger = getAllByRole('button', {
+      name: 'Open Sub-menu',
+    })[0];
```

Comment:
This seems brittle – if the university button is no longer the first in the menu, I wouldn't expect a test to fail. I'd rather we use test ids here to avoid this failing for reasons that we aren't trying to test for.

If we wanted to test the order of the menu, we should do that explicitly elsewhere.

Reply:
We don't have a testId for this button

Reply:
Could we add one?

Reply:
Yes we can!

Reply:
Actually, not as straightforward as I would've thought, because I dont think that we have glyph components set up to spread props. I can include that change in this PR though, if you would like.

Reply:
Looks like it should be spreading props actually!

Reply:
Discussed offline: props weren't spreading in the template so I updated, but that's actually not the problem in this case. In order to add a test-id we'd have to update SubMenu itself, which would also require relying on an order of the submenus. I actually don't think this is as brittle as you think, because we wouldn't change the order of the items in the UserMenu without updating the tests but happy to discuss any alternative thoughts that you have.